<a href="https://colab.research.google.com/github/pingping-0606/Python_dataviz2024/blob/main/Exam_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [4]:
 Data_Docter = pd.read_csv('/content/Data_Docter_ศ.นพ.ชวลิต.csv')

In [5]:
Data_Docter

name  \
0     ศ.นพ.ชวลิต ศูนย์มะเร็ง รพ.จุฬาลงกรณ์   
1                           Nubeer Kithiya   
2                          วาสนา คุ้มจิตต์   
3                     Wansook Chompoo Ploy   
4                              Aphit Aphit   
...                                    ...   
1648                           Fear Q'bick   
1649                          Toei Nittiya   
1650                       Ouiaii Promsuan   
1651                      Thanyalak Banboo   
1652                         Maii Mutchima   

                                               comments  ยอดไลค์  \
0     กดที่ลิงค์นี้เพื่อเข้าไปอ่านคอมเม้นของโพสต์ครั...      456   
1     เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พ...     1400   
2     แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
3     #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   
4     ลูกสาวเป็นมะเร็งกระดูก Osteosarcomaช่วงเข่า อา...      480   
...                                                 ...      ...   
1648            Jitladda Thonglukkeaw มาแชเร้ว ได้บุนนะ        0   
1649                                     Kamta Auttakan        0   
1650                                        อนุชา มีลาภ        0   
1651  Na Nat Fari Kawa II Herbara Binti Abdullah Ras...        0   
1652                              Sasipha Toomsoongnern        0   

      จำนวนการตอบกลับ  
0                   3  
1                  70  
2                  11  
3                  44  
4                   5  
...               ...  
1648                0  
1649                0  
1650                0  
1651                0  
1652                1  

[1653 rows x 4 columns]

จุดที่น่าสนใจของไฟล์ข้อมูลนี้นะคะ จุดแรกที่สนใจคือสนใจคอมเมนต์ที่มียอดไลค์
และการตอบกลับมากที่สุดค่ะเพราะถ้ามียอดไลค์ที่เยอะและคนตอบกลับหลายคนก็แปลว่าคอมเมนต์นั้นมีความน่าสนใจมากค่ะ

กรณีแรกนะคะเราจะมาดูที่ยอดไลค์ไลค์ โดยจะกำหนดยอดไลค์ที่มียอด 1000 ขึ้นไปค่ะ

In [23]:
Data_Docter[(Data_Docter['ยอดไลค์'] >= 1000)]

name                                           comments  \
1        Nubeer Kithiya  เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พ...   
3  Wansook Chompoo Ploy  #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...   
5            Ize Atitya  ลูกสาว 5 ขวบ เป็นมะเร็งสมองคะ ไม่มีอาการอะไร น...   

   ยอดไลค์  จำนวนการตอบกลับ  
1     1400               70  
3     1600               44  
5     1500               23

In [24]:
Data_Docter[(Data_Docter['ยอดไลค์'] >= 1000)].shape[0]

3

คอมเมนต์ของคุณ Wansook Chompoo Ploy มียอดไลค์สูงที่สุดคือ 1600 ยอดไลค์ คือเขาได้บอกว่าเขาได้ป่วยเป็นมะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอายุ 30... ตรวจpep.ทุกปี ไม่มีอะไรผิดปกติ วันดีคืนเดือด อยู่ๆภูมิแพ้ผิวหนังขึ้นทั้งวันทั้งคืน กินยายังไงก็ไม่ลง ทั้งๆที่ก่อนหน้านั้นก็รักษาภูมิแพ้ผิวหนังจนหายแล้ว เลยนึกขึ้นได้ว่า แพทย์ผิงหนังเคยเตือนว่า หากมีอาการภูมิแพ้แบบนี้อีก ให้ตรวจอย่างอื่นที่ผิดปกติร่วมด้วย ซึ่งมันคืออาการภูมิคุ้มกันบกพร่อง(ภูมิตก)แล้วแสดงสัญญาณทางผิวหนัง .... จึงไปตรวจpep.แบบคัดเชื้อhpvร่วมด้วย ณ วันที่หมออ่านผลคือ CIN ll ระยะเวลารอทำcolpo. 1เดือน ผลcolpo.(การส่องกล้องปากมดลูกด้วยคอลโปสโคป ) ขยับไป CIN lll หมอสั่งทำ Leep ทันที(เป็นการผ่าตัดโดยใช้ห่วงลวดไฟฟ้าในการตัดปากมดลูกเป็นรูปกรวย เพื่อเป็นการวินิจฉัยและรักษารอยโรคที่ปากมดลูกที่ผิดปกติระยะก่อนมะเร็ง ) เพราะระยะเดินเร็วเกินไป
ส่วนตัวนะคะ คิดว่าที่คุณคนนี้ได้ยอดไลค์เยอะ คงเป็นเพราะว่าเขาเคยเป็นภูมิแพ้ผิวหนัง แล้วบังเอิญนึกขึ้นได้ว่ามีแพทย์ผิวหนังเคยเตือนว่า หากมีอาการอีกให้ไปตรวจอย่างอื่นร่วมด้วย เลยทำให้เป็นอุทาหรณ์เตือนคนอื่นๆไปด้วยค่ะ ทำให้มีผู้คนสนใจกดไลค์ถึง 1600 ไลค์ค่ะ

กรณีที่ 2 นะคะ จะเป็นกรณีที่มีการตอบกลับมากที่สุดค่ะ โดยจะกำหนดเป็นคอมเมนต์ที่มีการตอบกลับ 30 คนขึ้นไปค่ะ

In [15]:
Data_Docter[(Data_Docter['จำนวนการตอบกลับ'] >= 30)]

name  \
1                Nubeer Kithiya   
3          Wansook Chompoo Ploy   
22  รมิตา ผู้หญิงที่เป็นของขวัญ   
46                Neung Lertsak   

                                             comments  ยอดไลค์  \
1   เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พ...     1400   
3   #มะเร็งปากมดลูกระยะCINlll เกือบเข้าระยะ 1 ตอนอ...     1600   
22  สำหรับมะเร็งปากมดลูก แนะนำให้ตรวจภายในเป็นประจ...      102   
46  ผมเป็นมะเร็งเม็ดสีผิวหนังชนิดร้ายแรง(Melanoma)...      477   

    จำนวนการตอบกลับ  
1                70  
3                44  
22               31  
46               30

In [25]:
Data_Docter[(Data_Docter['จำนวนการตอบกลับ'] >= 30)].shape[0]

4

ส่วนคอมเมนต์ที่มีการตอบกลับมากที่สุดนะคะคือคอมเมนต์ของคุณ Nubeer Kithiya มีการตอบกลับอยู่ที่ 70 การตอบกลับนะคะ โดยเขาบอกว่า เป็นมะเร็งลำไส้ใหญ่ ระยะที่ 3 ตอนอายุ 26 ค่ะ พฤติกรรมที่คิดว่ามีปัญหาคือ เลือกห้องน้ำที่จะขับถ่าย ถ้าไม่ใช่บ้าน หรือที่คุ้นเคย จะไม่ยอมถ่ายเลย จะอั้นไว้ ส่วนอาหารการกิน เราก็กินปกติ แบบคนทั่วไป ตามสั่ง ส้มตำ หมูทะ ชาบูบ้างตามโอกาส
เคสของคุณคนนี้มีการตอบกลับมากถึง 70 คอมเม้น คงเป็นเพราะว่าการขับถ่ายเป็นเรื่องใกล้ตัวมาก บวกกับเจ้าของคอมเมนต์อายุยังน้อย แต่ป่วยเป็นมะเร็งลำไส้ใหญ่ในวัยแค่ 26 ปี ทำให้เคสนี้เป็นอีกเคสหนึ่งที่น่าสนใจค่ะ

จากทั้งสองกรณีที่ยกตัวอย่างมา คอมเมนต์ของคุณ Nubeer Kithiya ติดอันดับทั้งสองกรณีเลย ถือว่าเป็นเคสที่ผู้คนในความสนใจมากที่สุดค่ะ คือ 1400 ยอดไลค์ และ 70 การตอบกลับ

จุดที่น่าสนใจจุดที่สองนะคะคือต้องการทราบว่าคนที่มาคอมเมนต์ผู้ที่ป่วยเป็นมะเร็งเกิดในผู้หญิงหรือผู้ชายมากกว่ากัน หนูเลยทำการหาคอมเมนต์ที่มีคำว่าพ่อและแม่ค่ะ โดยพ่อก็คือเพศชาย และแม่ก็คือเพศหญิงค่ะ

คอมเมนต์ที่มีคำว่า แม่

In [14]:
Data_Docter['comments'] = Data_Docter['comments'].astype(str)
count = Data_Docter['comments'].str.contains('แม่').sum()
print(count)


161


In [10]:
Data_Docter[Data_Docter['comments'].str.contains('แม่')]


name  \
2               วาสนา คุ้มจิตต์   
6                Pam Prechanont   
8     Varatchaya Thavanpongloed   
11                      Puy Puy   
15       Pariyaphat K Panumphun   
...                         ...   
843           Raphat Sangeumram   
847                   Oi Yippee   
854                เรณูณ่า ทิวา   
856                   Pooja Nip   
1639                  Amika Phs   

                                               comments  ยอดไลค์  \
2     แม่เป็นมะเร็งขั้วปอด ลามขึ้นสมองค่ะ เริ่มแรกที...      402   
6     คุณแม่เป็นมะเร็งเม็ดเลือดขาวแบบเฉียบพลันค่ะ คุ...      300   
8     แม่ตรวจเจอมะเร็งปี2554 ไม่มีอาการเจ็บปวดใดๆ แต...        0   
11    แม่แฟนเป็นมะเร็งตับอ่อน+ท่อน้ำดีค่ะ ไม่มีอาการ...       42   
15    แม่มีปีกจมูกที่บวมแต่แข็ง1ข้าง เป็นมานาน ไม่เจ...        0   
...                                                 ...      ...   
843   คุณแม่อายุ 60 ปีบริบูรณ์พอดี แล้วตรวจเจอ #มะเร...        0   
847   #คุณแม่คะเป็นมะเร็งลำไส้ ก่อนจะเป็นปี63แม่เป็น...        0   
854   แม่ตรวจเจอเป็นมะเร็งลำใส้ใหญ่ ระยะที่ 3 ค่ะ อา...        0   
856   ที่บ้านมี 2 เคสค่าคุณแม่ปี2553 มะเร็งลำไส้ใหญ่...        0   
1639  แม่เป็นมะเร็งตับค่ะ เพราะก่อนหน้านั้นแม่ทำงานร...        0   

      จำนวนการตอบกลับ  
2                  11  
6                   9  
8                   0  
11                  4  
15                  0  
...               ...  
843                 0  
847                 0  
854                 0  
856                 0  
1639                0  

[161 rows x 4 columns]

หาคอมเมนต์ที่มีคำว่า พ่อ

In [12]:
Data_Docter[Data_Docter['comments'].str.contains('พ่อ')].shape[0]


139

In [11]:
Data_Docter[Data_Docter['comments'].str.contains('พ่อ')]


name  \
10                นู๋นิด ขนิษฐา   
13    Nareerat Waawaaw Singtong   
17                Jin Ping Jern   
19                   WieWi Maka   
23                Aorry Salutuu   
...                         ...   
846             Mink Vetcharoen   
849            ฐิติพร จันทะบุตร   
1638            Zom O Thanyanun   
1643            Narisa Mohthong   
1644              Chanisara Hom   

                                               comments  ยอดไลค์  \
10    ขอให้เรื่องราวของพ่อกับพี่ชายเราเป็นวิทยาทานแก...      353   
13    คุณพ่อเปนมะเรงตับอ่อน+ท่อน้ำดี ไม่มีอาการอะไรเ...      291   
17    คุณพ่อเราเป็น #มะเร็งหลังโพรงจมูก ค่ะ นับย้อนไ...      210   
19    พ่อเป็นมะเร็งกระเพาะอาหาร พ่อไม่สูบบุหรี่ แต่พ...        0   
23    พ่อเป็นมะเร็งเม็ดเลือดขาว…รักษาอยู่ประมาน3ปี พ...        0   
...                                                 ...      ...   
846   พ่อเป็นมะเร็งปอดค่ะ(มะเร็งกินกระดูกซี่โครงขวาท...        0   
849   พ่อผมเริ่มจากเจ็บท้อง ผอมลงเรื่อยๆ ตรวจ3-4 รพ....        0   
1638  คุณพ่อเป็นมะเร็งปอด  สูบบุหรี่มา 40ปี + ร่างกา...        0   
1643  พ่อเป็น มะเร็งปอด ระยะ4 (ยังไม่แพร่กระจาย)แบบไ...        0   
1644  พ่อเป็นมะเร็งลำไส้ชอบกินอาหารสุกๆ ดิบค่ะ ขับถ่...        0   

      จำนวนการตอบกลับ  
10                 25  
13                 21  
17                  3  
19                  0  
23                  0  
...               ...  
846                 0  
849                 0  
1638                0  
1643                0  
1644                0  

[139 rows x 4 columns]

จากข้อมูลที่ได้ออกมานะคะ คอมเมนต์ที่มีคำว่าแม่มีทั้งหมด 161 คอมเมนต์ และคอมเมนต์ที่มีคำว่าพ่อมีทั้งหมด 139 คอมเมนต์ค่ะ สรุปได้ว่าจากคอมเมนต์ทั้งหมด มีผู้หญิงป่วยเป็นมะเร็งมากกว่าผู้ชายค่ะ